src/main/java/apoc/convert/Convert.java

```diff
@@ -91,48 +91,69 @@ else if (list.getClass().isArray() && !list.getClass().getComponentType().isPrim
         return list == null ? null : new SetBackedList(new LinkedHashSet<>(list));
     }
     
-    @SuppressWarnings("rawtypes")
 	@UserFunction
-    @Description("apoc.convert.toListOf(value, type) | tries it's best to convert "
-    		+ "the value to a list of supported type, supported types are: "
-    		+ "int, long, double, String, boolean, node, relationship")
-    public List toListOf(@Name("list") Object list, @Name("type") String type) {
-        List objectList = convertToList(list);
+    @Description("apoc.convert.toIntList(value) | tries it's best to convert "
+    		+ "the value to a list of integers")
+    public List<Integer> toIntList(@Name("list") Object list) {
```

Comment:
List<Long> as Cypher doesn't know Integers but calls it Integer

and use toInteger function